subsys/bluetooth/host/ecc.c

```diff
@@ -180,11 +199,13 @@ void bt_hci_evt_le_pkey_complete(struct net_buf *buf)
 		atomic_set_bit(bt_dev.flags, BT_DEV_HAS_PUB_KEY);
 	}
 
-	for (cb = pub_key_cb; cb; cb = cb->_next) {
-		cb->func(evt->status ? NULL : pub_key);
+	SYS_SLIST_FOR_EACH_CONTAINER(&pub_key_cb_slist, cb, node) {
+		if (cb->func) {
+			cb->func(evt->status ? NULL : pub_key);
+		}
 	}
 
-	pub_key_cb = NULL;
+	sys_slist_init(&pub_key_cb_slist);
```

Comment:
aren't callbacks suppose to be preserved?    (calling bt_pub_key_gen(NULL) is allowed according to docs to refresh keys)

Reply:
Maybe update the documentation to match the new behavior then? There are only users of this API, as long as both work with the current behavior that should be fine.

Reply:
yeap, lets keep this as is and update docs